src/bp/ui-studio/src/web/views/Config/index.tsx

```diff
@@ -0,0 +1,507 @@
+  }
+
+  async componentDidMount() {
+    const languages = await this.fetchLanguages()
+    const licensing = await this.fetchLicensing()
+
+    if (!this.props.bot) {
+      this.props.fetchBotInformation()
+    }
+
+    const bot = this.props.bot
+    const status = bot.disabled ? 'disabled' : bot.private ? 'private' : 'public'
+
+    this.initialFormState = {
+      name: bot.name || '',
+      status: this.state.statuses.find(s => s.value === status),
+      description: bot.description || '',
+      selectedDefaultLang: languages.find(l => l.value === bot.defaultLanguage),
+      selectedLanguages: languages.filter(x => bot.languages.includes(x.value)),
+      website: bot.details.website || '',
+      phoneNumber: bot.details.phoneNumber || '',
+      emailAddress: bot.details.emailAddress || '',
+      termsConditions: bot.details.termsConditions || '',
+      privacyPolicy: bot.details.privacyPolicy || '',
+      avatarUrl: bot.details.avatarUrl || '',
+      coverPictureUrl: bot.details.coverPictureUrl || ''
+    }
+
+    this.setState({
+      ...this.initialFormState,
+      licensing,
+      languages
+    })
+  }
+
+  async fetchLanguages(): Promise<SelectItem[]> {
+    const { data } = await axios.get('admin/languages/available', axiosConfig)
+    const languages = _.sortBy(data.languages, 'name').map(lang => ({
+      label: lang.name,
+      value: lang.code
+    }))
+    return languages
+  }
+
+  async fetchLicensing(): Promise<Licensing> {
+    const { data } = await axios.get('admin/license/status', axiosConfig)
+    return data.payload
+  }
+
+  saveChanges = async () => {
+    this.setState({ error: undefined, isSaving: true })
+
+    const bot: BotConfig = {
+      name: this.state.name,
+      disabled: this.state.status.value === 'disabled',
+      private: this.state.status.value === 'private',
+      description: this.state.description,
+      defaultLanguage: this.state.selectedDefaultLang.value,
+      languages: this.state.selectedLanguages.map(x => x.value),
+      details: {
+        website: this.state.website,
+        phoneNumber: this.state.phoneNumber,
+        emailAddress: this.state.emailAddress,
+        termsConditions: this.state.termsConditions,
+        privacyPolicy: this.state.privacyPolicy,
+        avatarUrl: this.state.avatarUrl,
+        coverPictureUrl: this.state.coverPictureUrl
+      }
+    }
+
+    const { error } = Joi.validate(bot, BotEditSchema)
+    if (error) {
+      toastFailure('The form contains errors')
+      this.setState({ error: error, isSaving: false })
+      return
+    }
+
+    try {
+      const disableChanged = bot.disabled != (this.initialFormState.status.value === 'disabled')
+      let allow = true
+
+      if (disableChanged && bot.disabled) {
+        allow = await confirmDialog(
+          `Are you sure want to unmount this bot? All of the functionalities of this bot will become unavailable.`,
+          { acceptLabel: 'Unmount' }
+        )
+      }
```

Comment:
I'm not sure if that additional dialog is necessary, the user is already choosing "Disabled" in the list, then clicking "Save changes" . He can't misclick or choose this by mistaked